pandas/tests/series/indexing/test_multiindex.py

```diff
@@ -20,3 +21,13 @@ def test_access_none_value_in_multiindex():
     s = Series([1] * len(midx), dtype=object, index=midx)
     result = s.loc[("Level1", "Level2_a")]
     assert result == 1
+
+
+def test_nat_multi_index():
+    # GH36541: that reset_index() does not raise ValueError
+    ix = pd.MultiIndex.from_tuples([(pd.NaT, 1), (pd.NaT, 2)], names=["a", "b"])
```

Comment:
can you try this with one element a Timestamp (and one a NaT)
also add a case where you have a Timedelta and a NaT (and I think we might need to change the code slightly)

can parameterize the test.

Reply:
added test cases. please have another look and advise.